addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java

```diff
@@ -0,0 +1,178 @@
+
+    protected BigDecimal tist;
+    protected BigDecimal tsoll;
+    protected BigDecimal absenk;
+    protected BigDecimal komfort;
+    protected BigDecimal lock;
+    protected BigDecimal devicelock;
+    protected String errorcode;
+    protected BigDecimal batterylow;
+    protected Nextchange nextchange;
+
+    public BigDecimal getTist() {
+        return tist != null ? tist.multiply(TEMP_FACTOR) : BigDecimal.ZERO;
+    }
+
+    public void setTist(BigDecimal tist) {
+        this.tist = tist;
+    }
+
+    public BigDecimal getTsoll() {
+        if (tsoll == null) {
+            return BigDecimal.ZERO;
+        } else if (tsoll.compareTo(TEMP_ON) == 0) {
+            return TEMP_MAX.add(new BigDecimal("2.0"));
+        } else if (tsoll.compareTo(TEMP_OFF) == 0) {
+            return TEMP_MIN.subtract(new BigDecimal("2.0"));
+        } else {
+            return tsoll.multiply(TEMP_FACTOR);
+        }
+    }
+
+    public void setTsoll(BigDecimal tsoll) {
+        this.tsoll = tsoll;
+    }
+
+    public BigDecimal getKomfort() {
+        return komfort != null ? komfort.multiply(TEMP_FACTOR) : BigDecimal.ZERO;
+    }
+
+    public void setKomfort(BigDecimal komfort) {
+        this.komfort = komfort;
+    }
+
+    public BigDecimal getAbsenk() {
+        return absenk != null ? absenk.multiply(TEMP_FACTOR) : BigDecimal.ZERO;
+    }
+
+    public void setAbsenk(BigDecimal absenk) {
+        this.absenk = absenk;
+    }
+
+    public BigDecimal getLock() {
+        return lock;
+    }
+
+    public void setLock(BigDecimal lock) {
+        this.lock = lock;
+    }
+
+    public BigDecimal getDevicelock() {
+        return devicelock;
+    }
+
+    public void setDevicelock(BigDecimal devicelock) {
+        this.devicelock = devicelock;
+    }
+
+    public String getErrorcode() {
+        return errorcode;
+    }
+
+    public void setErrorcode(String errorcode) {
+        this.errorcode = errorcode;
+    }
+
+    public BigDecimal getBatterylow() {
+        return batterylow;
+    }
+
+    public void setBatterylow(BigDecimal batterylow) {
+        this.batterylow = batterylow;
+    }
+
+    public Nextchange getNextchange() {
+        return nextchange;
+    }
+
+    public void setNextchange(Nextchange value) {
+        this.nextchange = value;
+    }
+
+    public String toString() {
+        return new ToStringBuilder(this).append("tist", this.getTist()).append("tsoll", this.getTsoll())
+                .append("absenk", this.getAbsenk()).append("komfort", this.getKomfort()).append("lock", this.getLock())
+                .append("errorcode", this.getErrorcode()).append("batterylow", this.getBatterylow())
+                .append("nextchange", this.getNextchange()).toString();
+    }
+
+    @XmlType(name = "", propOrder = { "endperiod", "tchange" })
+    public static class Nextchange {
+
+        protected int endperiod;
+        protected BigDecimal tchange;
+
+        /**
+         * Ruft den Wert der endperiod-Eigenschaft ab.
```

Comment:
Did you generate this file or create it yourself? I like the comments to be English, it takes me less time to read them ;-)